Horizontal tick label fix

rbojha opened the pull request:
HI, We were using `bootstrap-slider` for our project and found this issue, where when you resize the window the tick labels are not able update its new position with respect to the new resized window.

I tried to solve it with the help of using % width for labels instead of size/ (no of ticks) , the size is available in state.

Here are the two fiddles which shows the problem and solution (may be `approx`)

Fiddle, with original code:
https://jsfiddle.net/raviojha/g1cm1gsj/

Fiddle with updated code, with slide correction  :
https://jsfiddle.net/raviojha/b8b1gu5f/1/

Note: you need to resize the window (or result window) (horizontally) to see the difference.

seiyria commented:
Please rebase out your unnecessary commits, add appropriate unit tests, and make it pass the build. Thanks!

rbojha commented:
@seiyria , can you please take a look at this??

seiyria commented:
It's still missing a unit test, but that diff looks a _lot_ better!

rbojha commented:
Thanks @seiyria , let me work on that...

rbojha commented:
HI @seiyria , can we look at this, 
Please suggest.

seiyria commented:
This looks ok, but I'm going to have to defer to @rovolution to do a final review.

rovolution commented on js/bootstrap-slider.js:
```diff
@@ -984,20 +984,15 @@
 					var minTickValue = Math.min.apply(Math, this.options.ticks);
 
 					var styleSize = this.options.orientation === 'vertical' ? 'height' : 'width';
-					var styleMargin = this.options.orientation === 'vertical' ? 'marginTop' : 'marginLeft';
```
why is this code being removed? how does this affect the orientation when the slider is vertical instead of horizontal?

id like to see a fiddle for this

rovolution commented on js/bootstrap-slider.js:
```diff
@@ -984,20 +984,15 @@
-									extraMargin = this.tickLabelContainer.childNodes[i].offsetHeight;
-								}
+						/* Chidren are position absolute, calculate height by finding the max offsetHeight of a child */
+						for (i = 0 ; i < this.tickLabelContainer.childNodes.length; i++) {
```
need to scope `i` by declaring it with `var i`

rovolution commented on js/bootstrap-slider.js:
```diff
@@ -1024,13 +1019,19 @@
+
+							this.tickLabels[i].style.position = 'absolute';
+							if (this.options.orientation === 'horizontal') {
+				                this.tickLabels[i].style['transform'] = 'translate(-50%, 0%)';
```
could you fix the spacing on this?

Bhoft replied:
One question why are you doing the translate in JS and not in CSS? (as i have done in my fork)
```
.slider.slider-vertical .slider-tick-label {
    padding-left: 4px;
    -moz-transform: translateX(50%) translateY(-50%);
    -webkit-transform: translateX(50%) translateY(-50%);
    transform: translateX(50%) translateY(-50%);
}

.slider.slider-horizontal .slider-tick-label {
    -moz-transform: translateX(-50%) translateY(0%);
    -webkit-transform: translateX(-50%) translateY(0%);
    transform: translateX(-50%) translateY(0%);
}
````

rovolution replied:
@rbojha could you apply this transform via LESS instead?

rovolution commented on test/specs/WindowResizeTickLabelSpec.js:
```diff
@@ -0,0 +1,47 @@
+            $("div.slider").width($(window).width());
+        };
+
+        expect($el.siblings('div.slider').find('.slider-tick-label:eq(0)').width()).toBe($el.siblings('div.slider').width()*per);
```
:+1: for this test!

rovolution commented:
@rbojha see my comments in the source code. overall it looked solid

rbojha commented:
@rovolution, Please find below the updated fiddle, 

Fiddle, with original code:
http://jsfiddle.net/raviojha/g1cm1gsj/4/

Fiddle with updated code, with slight correction :
http://jsfiddle.net/raviojha/b8b1gu5f/5/

The `styleMargin` is removed which is adding top/left `negative` margin to the tick labels which is half of its width which is taken care by adding transition.

rovolution commented:
@rbojha could you please rebase? thanks!

zsiswick commented:
I'd like to get this fix merged in to master, how can I help?

seiyria commented:
@zsiswick Really all that needs to be done is aggregate all of the commits here, rebase them, and make sure it's mergable against master. If all of that is true then it can be merged quickly.

rbojha commented:
Hey Guys, i have rebased it but unfortunately some other test cases are failing, must be some logical mis-understanding from my end, i will work on this on coming weekend, for the time being i have updated jsfiddle with the new and old code (latest code i mean from main repo), for you guys to look at it....

#### Using Original Code
http://jsfiddle.net/raviojha/b8ykw84e/1/

#### Using Tweaked Code
http://jsfiddle.net/raviojha/7uk407cp/2/

Apologies again for replying late
